src/main/java/edu/kit/kastel/sdq/artemis4j/client/ProgrammingSubmissionDTO.java

```diff
@@ -51,9 +51,19 @@ public static void saveAssessment(ArtemisClient client, long participationId, bo
         ArtemisRequest.put().path(List.of("participations", participationId, "manual-results")).param("submit", submit).body(result).execute(client);
     }
 
+    public List<ResultDTO> nonAutomaticResults() {
+        return this.results().stream().filter(r -> r.assessmentType() != AssessmentType.AUTOMATIC).toList();
+    }
+
     private ProgrammingSubmissionDTO fetchLongFeedback(ArtemisClient client) throws ArtemisNetworkException {
         List<ResultDTO> results = new ArrayList<>(this.results().size());
         for (var result : this.results()) {
+            // sometimes the feedbacks are not loaded, to fetch the long feedbacks, we need
+            // to load the feedbacks first
+            if (result.feedbacks() == null) {
+                result = result.fetchFeedbacks(client, this.participation().id());
+            }
+
```

Comment:
Do you have an example where this is actually necessary (and we are not just using the wrong API endpoint for a certain task)?

Reply:
This is necessary, because of the current artemis4j implementation. It will always call the fetchlongfeedbacks for every result in a submission response. The method itself will remove all results that do not have feedback, so my submission will never have any results.

Without the results (the ids) I am unable to call the endpoint of the above code.

Reply:
The least intrusive option was to fetch the feedback when it is missing. It seems like the feedback is present under some conditions, so I would like to make Artemis4J handle the fetching of the feedback (if necessary). 

Similar to long feedbacks, the user doesn't want to manually fetch the feedbacks. They might not even expect them to be missing, like it happened to me. Took me a while to figure out that my results are all empty

Reply:
One idea that comes to mind, is to only fetch the long feedback/feedbacks if they are accessed. I assume the LazyNetworkValue could be used for that?

Reply:
Ok, so it looks like the exercises/id/programming-submissions endpoint just never returns feedbacks. I guess the best solution would be to just remove long feedback fetching from ProgrammingSubmissionDTO::fetchAll, and implement a `fetchDetails(participationId, resultId)` method in `ResultDTO` (that fetches long feedbacks), similar to the old Artemis4J implementation.